db/init.sql

```diff
@@ -4,13 +4,315 @@ CREATE DATABASE mm_db;
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
 CREATE TYPE user_role AS ENUM ('admin', 'teacher', 'student');
+CREATE TYPE attempt_state AS ENUM ('in_progress', 'completed');
+-- NOTE(nrydanov): Need to think of other types together
+CREATE TYPE block_type AS ENUM ('assignment', 'text', 'media', 'quiz');
+
 
 CREATE TABLE users (
     id uuid NOT NULL PRIMARY KEY,
+    -- Auth data
     email varchar(128) UNIQUE NOT NULL,
+    password bytea NOT NULL,
+    -- Identity data
     name varchar(32) NOT NULL,
     surname varchar(32) NOT NULL,
     patronymic varchar(32),
-    role user_role NOT NULL,
-    password bytea NOT NULL
+    date_of_birth date,
+    
+    -- University data
+    -- NOTE(nrydanov): This data should be enough to determine exact academic group
+    faculty_id uuid NOT NULL,
```

Comment:
Shouldn't we also store the university ID?

Reply:
If we assume that our product will be self-hosted by every university, then it should be there. But by adding this field we can provide more flexibility, so I think it's good point.